python-threatexchange/threatexchange/cli/experimental_fetch.py

```diff
@@ -36,75 +44,102 @@ def init_argparse(cls, ap) -> None:
-        dataset.load_indicator_cache(self.signal_types_by_name.values())
-        more_to_fetch = True
+        request_time = int(time.time())
+        privacy_group = dataset.config.privacy_groups[0]
+        self.indicator_store = dataset.get_indicator_store(privacy_group)
+        dataset.load_indicator_cache(self.indicator_store)
+        self.stop_time = request_time if self.stop_time is None else self.stop_time
         next_page = None
+        if self.continuation:
+            checkpoint = dataset.get_indicator_checkpoint(privacy_group)
+            self.start_time = checkpoint["last_stop_time"]
+            self.stop_time = request_time
+            self.threat_types = checkpoint["threat_types"]
+            next_page = checkpoint["url"]
+            if request_time - checkpoint["last_run_time"] > self.DEFAULT_REFETCH_SEC:
+                print("It's been a long time since a full fetch, forcing one now.")
+                self.start_time = 0
+                next_page = None
 
+        more_to_fetch = True
+        remaining_attempts = 5
         while more_to_fetch:
-            result = TE.Net.getThreatUpdates(
-                dataset.config.privacy_groups[0],
-                start_time=self.start_time,
-                stop_time=self.stop_time,
-                owner=self.owner,
-                threat_type=self.threat_types,
-                additional_tags=self.additional_tags,
-                next_page=next_page,
-            )
-            if "data" in result:
-                self._process_indicators(result["data"])
-            more_to_fetch = "paging" in result and "next" in result["paging"]
-            next_page = result["paging"]["next"] if more_to_fetch else None
+            try:
+                result = TE.Net.getThreatUpdates(
+                    privacy_group,
+                    start_time=self.start_time,
+                    stop_time=self.stop_time,
+                    threat_type=self.threat_types,
+                    next_page=next_page,
+                    limit=self.limit,
+                )
+                if "data" in result:
+                    self._process_indicators(result["data"])
+                more_to_fetch = "paging" in result and "next" in result["paging"]
+                next_page = result["paging"]["next"] if more_to_fetch else ""
+            except Exception as e:
+                remaining_attempts -= 1
+                print(f"The following error occured:\n{e}")
+                if type(e) is urllib.error.HTTPError:
+                    print(e.read())
+                if remaining_attempts > 0:
+                    print(f"\nTrying again {remaining_attempts} more times.")
+                    time.sleep(5)
+                    continue
+                else:
+                    print(
+                        "\n5 consecutive errors occured, saving state and shutting down!"
+                    )
+                    break
+            remaining_attempts = 5
```

Comment:
Yup, 5 was just an arbitrary number I came up with - this is just to make sure that if there is some TAO transient error we don't stop querying and instead try again which most of the time fixes things, but if after 5 attempts you're still running into errors it will checkpoint whats left to do and a continuation can be manually started using the continuation option

Something to note is I haven't actually hit any non intentional errors (aside form when my internet went down), so overall I think this is pretty stable

Reply:
(not blocking): I think that this type of retry will probably be fragile, and the root problem is we need to switch to using the `requests` module in the netcode rather than handrolling with python less-powerful default libs

Reply:
I initially wrote the first version of experimentalfetch using `requests` but IIRC it was asked that I switch back to using the existing way of doing things, unfortunately I don't have the time to update this atm but would you like me to leave a TODO?